extensions-contrib/druid-deltalake-extensions/src/main/java/org/apache/druid/delta/input/DeltaInputRow.java

```diff
@@ -0,0 +1,219 @@
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(row, schema, fieldNameToOrdinal, delegateRow);
+  }
+
+  @Override
+  public String toString()
+  {
+    return "DeltaInputRow{" +
+           "row=" + row +
+           ", schema=" + schema +
+           ", fieldNameToOrdinal=" + fieldNameToOrdinal +
+           ", delegateRow=" + delegateRow +
+           '}';
+  }
+
+  public Map<String, Object> getRawRowAsMap()
+  {
+    return RowSerde.convertRowToJsonObject(row);
+  }
+
+  @Nullable
+  private Object _getRaw(String dimension)
+  {
+    StructField field = schema.get(dimension);
+    if (field == null || field.isMetadataColumn()) {
+      return null;
+    }
+
+    int ordinal = fieldNameToOrdinal.getInt(dimension);
+    if (ordinal < 0) {
+      return null;
+    }
+    return getValue(field.getDataType(), row, ordinal);
+  }
+
+  @Nullable
+  private static Object getValue(DataType dataType, io.delta.kernel.data.Row dataRow, int columnOrdinal)
+  {
+    if (dataRow.isNullAt(columnOrdinal)) {
+      return null;
+    } else if (dataType instanceof BooleanType) {
+      return dataRow.getBoolean(columnOrdinal);
+    } else if (dataType instanceof ByteType) {
+      return dataRow.getByte(columnOrdinal);
+    } else if (dataType instanceof ShortType) {
+      return dataRow.getShort(columnOrdinal);
+    } else if (dataType instanceof IntegerType) {
+      return dataRow.getInt(columnOrdinal);
+    } else if (dataType instanceof DateType) {
+      // DateType data is stored internally as the number of days since 1970-01-01
+      int daysSinceEpochUTC = dataRow.getInt(columnOrdinal);
+      return LocalDate.ofEpochDay(daysSinceEpochUTC).atStartOfDay(ZONE_ID).toEpochSecond();
+    } else if (dataType instanceof LongType) {
+      return dataRow.getLong(columnOrdinal);
+    } else if (dataType instanceof TimestampType) {
+      // TimestampType data is stored internally as the number of microseconds since epoch
+      long microSecsSinceEpochUTC = dataRow.getLong(columnOrdinal);
+      LocalDateTime dateTime = LocalDateTime.ofEpochSecond(
+          microSecsSinceEpochUTC / 1_000_000 /* epochSecond */,
+          (int) (1000 * microSecsSinceEpochUTC % 1_000_000) /* nanoOfSecond */,
+          ZoneOffset.UTC
+      );
+      return dateTime.atZone(ZONE_ID).toInstant().toEpochMilli();
+    } else if (dataType instanceof FloatType) {
+      return dataRow.getFloat(columnOrdinal);
+    } else if (dataType instanceof DoubleType) {
+      return dataRow.getDouble(columnOrdinal);
+    } else if (dataType instanceof StringType) {
+      return dataRow.getString(columnOrdinal);
+    } else if (dataType instanceof BinaryType) {
+      final byte[] arr = dataRow.getBinary(columnOrdinal);
+      final char[] charArray = new char[arr.length];
+      for (int i = 0; i < arr.length; i++) {
+        charArray[i] = (char) (arr[i] & 0xff);
+      }
+      return String.valueOf(charArray);
```

Comment:
can you describe the logic? why does a single byte need to be masked with `0xff`?

Reply:
Good catch, I was trying to convert it to an unsigned value, which is not needed in this case